lib/spack/docs/basic_usage.rst

```diff
@@ -1060,14 +1060,19 @@ set of arbitrary versions, such as ``@1.0,1.5,1.7`` (``1.0``, ``1.5``,
 or ``1.7``).  When you supply such a specifier to ``spack install``,
 it constrains the set of versions that Spack will install.
 
+For packages with a ``git`` attribute, a 40-character git commit sha
```

Comment:
More of a functional question here. Does it have to be the full 40 character sha?  The native git client allows for an abbreviated sha.  Will it throw if I give 12 characters?

Reply:
Currently it will throw if you give it 12 characters. That's another thing we want to improve as future work, but it made sense to get the feature in with the 40-char constraint first.